Changelog.md

```diff
@@ -14,6 +14,7 @@ Compiler Features:
 Bugfixes:
  * Commandline Interface: Disallow the following options outside of the compiler mode: ``--via-ir``,``--metadata-literal``, ``--metadata-hash``, ``--model-checker-show-unproved``, ``--model-checker-div-mod-no-slacks``, ``--model-checker-engine``, ``--model-checker-invariants``, ``--model-checker-solvers``, ``--model-checker-timeout``, ``--model-checker-contracts``, ``--model-checker-targets``.
  * Type Checker: Fix null dereference in `abi.encodeCall` type checking of free function.
+ * Type Checker: Fix out of range error for nested tuples.
```

Comment:
I think this should be a bit more specific. I'd need to take a closer look to give a proper suggestion but just at a quick glance, as a user, this does not tell me when I can run into this. this isn't caused by nesting alone, right? Does it specifically require an assignment or can it happen in other cases?

Also I'd say "compiler crash" rather than "out of range error". The specific type of an exception and even the fact that it's a C++ exception is just an unimportant implementation detail here in my opinion.

Reply:
Right. I'll think of a better way to phrase it. But yes it's really about nesting outer unary tuples during assignment.